app/models/v2/test_result.rb

```diff
@@ -70,6 +70,15 @@ class TestResult < ApplicationRecord
       { conditions: systems.arel.where_sql.gsub(/^where /i, '') }
     end
 
+    searchable_by :failed_rule_severity, %i[in] do |_key, _op, val|
+      values = val.split(',').map(&:strip)
+      ids = ::V2::TestResult.joins(rule_results: :rule)
+                            .where(v2_rules: { severity: values }, rule_results: { result: 'fail' })
+                            .reselect(:id)
+
+      { conditions: "v2_test_results.id IN (#{ids.to_sql})" }
```

Comment:
I know why you're bumping into the select cache: you don't need to select from the test results table at all. You can directly go to rule_results and join it with rules + select by `test_result_id`. That way you're joining one less table and you won't need the `reselect`